test/unit/range/view/view_istreambuf_test.cpp

```diff
@@ -0,0 +1,67 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/range/view/istreambuf.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/complement.hpp>
+#include <seqan3/std/ranges>
+
+#include <seqan3/core/debug_stream.hpp>
+
+using namespace seqan3;
+
+TEST(view_istreambuf, basic)
+{
+    std::string data{"ACGTATATATAT ATATAT TTA \n AUAUAA"};
+    std::istringstream is{data};
+
+    // construct from istream:
+    auto v1 = view::istreambuf(is);
+    size_t i = 0;
+    for (auto c : v1)
+        EXPECT_EQ(c, data[i++]);
+    EXPECT_EQ(i, data.size());
+
+    // construct from streambuf
+    is.clear();
+    is.seekg(0, std::ios::beg);
+    auto v2 = view::istreambuf(*is.rdbuf());
+    i = 0;
+    for (auto c : v2)
+        EXPECT_EQ(c, data[i++]);
+    EXPECT_EQ(i, data.size());
+
+    // combinability
+    is.clear();
+    is.seekg(0, std::ios::beg);
+    auto v3 = view::istreambuf(is) | view::char_to<dna5> | view::complement;
```

Comment:
can you test a take_until (instead) since it is the usual stuff we need in the io module?